kotlin/src/main/kotlin/org/jdbi/v3/core/kotlin/KotlinMapper.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.kotlin
+
+import org.jdbi.v3.core.mapper.RowMapper
+import org.jdbi.v3.core.statement.StatementContext
+import java.lang.reflect.InvocationTargetException
+import java.sql.ResultSet
+import java.sql.SQLException
+import kotlin.reflect.KClass
+import kotlin.reflect.KParameter
+import kotlin.reflect.jvm.isAccessible
+import kotlin.reflect.jvm.javaType
+import kotlin.reflect.primaryConstructor
+
+class KotlinMapper<C : Any> internal constructor(private val clazz: Class<C>) : RowMapper<C> {
+    private val kclass: KClass<C> = clazz.kotlin
+
+    @Throws(SQLException::class)
+    override fun map(rs: ResultSet, ctx: StatementContext): C {
+        try {
+            return tryMap(rs, ctx)
```

Comment:
Not a blocker: overriding the `RowMapper.specialize` method would help performance.

Specialized mappers are used on only one result set, so you can introspect types and match up column names to properties/parameters once and reuse it for the whole result set, instead of introspecting and matching on every row.

See `BeanMapper` for an example.